src/cascadia/LocalTests_TerminalApp/CommandlineTest.cpp

```diff
@@ -0,0 +1,802 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include <WexTestClass.h>
+
+#include "../TerminalApp/AppCommandlineArgs.h"
+
+using namespace WEX::Logging;
+using namespace WEX::Common;
+using namespace WEX::TestExecution;
+
+using namespace winrt::TerminalApp;
+using namespace ::TerminalApp;
+
+namespace TerminalAppLocalTests
+{
+    // TODO:microsoft/terminal#3838:
+    // Unfortunately, these tests _WILL NOT_ work in our CI. We're waiting for
+    // an updated TAEF that will let us install framework packages when the test
+    // package is deployed. Until then, these tests won't deploy in CI.
+    class CommandlineTest
+    {
+        // Use a custom AppxManifest to ensure that we can activate winrt types
+        // from our test. This property will tell taef to manually use this as
+        // the AppxManifest for this test class.
+        // This does not yet work for anything XAML-y. See TabTests.cpp for more
+        // details on that.
+        BEGIN_TEST_CLASS(CommandlineTest)
+            TEST_CLASS_PROPERTY(L"RunAs", L"UAP")
+            TEST_CLASS_PROPERTY(L"UAP:AppXManifest", L"TerminalApp.LocalTests.AppxManifest.xml")
+        END_TEST_CLASS()
+
+        TEST_METHOD(TryCreateWinRTType);
+
+        TEST_METHOD(ParseSimpleCommandline);
+        TEST_METHOD(ParseTrickyCommandlines);
+        TEST_METHOD(TestEscapeDelimiters);
+
+        TEST_METHOD(ParseSimpleHelp);
+        TEST_METHOD(ParseBadOptions);
+        TEST_METHOD(ParseSubcommandHelp);
+
+        TEST_METHOD(ParseBasicCommandlineIntoArgs);
+        TEST_METHOD(ParseNewTabCommand);
+        TEST_METHOD(ParseSplitPaneIntoArgs);
+        TEST_METHOD(ParseComboCommandlineIntoArgs);
+        TEST_METHOD(ParseFocusTabArgs);
+
+        TEST_METHOD(ParseNoCommandIsNewTab);
+

```

Comment:
Test excessively long text. Test empty text. Test text with ridiculous Unicode points in it like RTL/LTR/ZWJ/BOM. Test with invalid Unicode characters like the 0x80 0xc0 version of null.

Reply:
-> #4168